File: sdkwork-ai-sdk-java/src/main/java/com/sdkwork/ai/api/EmbeddingsApi.java
package com.sdkwork.ai.api;

import com.sdkwork.ai.http.HttpClient;
import com.sdkwork.ai.model.*;
import java.util.List;
import java.util.Map;

public class EmbeddingsApi {
    private final HttpClient client;
    
    public EmbeddingsApi(HttpClient client) {
        this.client = client;
    }

    /** Create embeddings */
    public EmbeddingResponse createEmbedding(EmbeddingRequest body) throws Exception {
        return (EmbeddingResponse) client.post(ApiPaths.aiPath("/embeddings"), body);
    }
}

File: sdkwork-ai-sdk-kotlin/src/main/kotlin/com/sdkwork/ai/api/EmbeddingsApi.kt
package com.sdkwork.ai.api

import com.sdkwork.ai.*
import com.sdkwork.ai.http.HttpClient

class EmbeddingsApi(private val client: HttpClient) {

    /** Create embeddings */
    suspend fun createEmbedding(body: EmbeddingRequest): EmbeddingResponse? {
        return client.post(ApiPaths.aiPath("/embeddings"), body) as? EmbeddingResponse
    }
}
